### PR TITLE
Doc 7956 htbt code samples

### PR DESCRIPTION
**Pull request by ibsoln**
https://issues.couchbase.com/browse/DOC-7956

DOC-7956 Replicator Heart Beat code samples

@ibsoln DOC-7956-htbt -- iOS code samples … cdeb3b3
@ibsoln DOC-7956-htbt -- Java and Android code samples … 570c8f1
@ibsoln DOC-7956-htbt -- C#/.Net code samples … dd35440

### CODE REVIEW

**Comment by ibsoln** [comment]
Can you please review the appropriate code samples for Replicator Heartbeat ?

- @jayahariv  iOS code samples … cdeb3b3
- @bmeike  -- Java and Android code samples … 570c8f1
- @Sandychuang8  -- C#/.Net code samples … dd35440

File: modules/objc/examples/code_snippets/SampleCodeTest.m [hunk 1]
```diff
@@ -934,6 +934,25 @@ - (void) dontTestReplicationPullFilter {
...
+//  Added 2/Feb/21 - Ian Bridge
+    - void dontTestCustomHeartbeat {
+    // tag::replication-set-heartbeat[]
+        id target =
+          [[CBLURLEndpoint alloc] initWithURL: [NSURL URLWithString: @"ws://foo.cbl.com/db"]];
+
+        CBLReplicatorConfiguration* config = [self configWithTarget: target
+                                                            type: kCBLReplicatorTypePush
+                                                      continuous: YES];
+
```
Comment (jayahariv) [inline modules/objc/examples/code_snippets/SampleCodeTest.m]: can we use simple initialization for the ReplicatorConfiguration like below. The above sample function(`configWithTarget`), is internal function, if users copy paste the above, they might see build errors. 

```
CBLReplicatorConfiguration* config = [[CBLReplicatorConfiguration alloc] initWithDatabase: db target: target];
```

File: modules/swift/examples/code_snippets/SampleCodeTest.swift [hunk 2]
```diff
@@ -779,6 +779,23 @@ class SampleCodeTest {
...
+    func testCustomHeartbeat() {
+        // tag::replication-set-heartbeat[]
+        let target =
+          URLEndpoint(url: URL(string: "ws://foo.couchbase.com/db")!)
+
+        let config =
+          self.config(target: target,
+            type: .pushAndPull,
+            continuous: true)
+
```
Comment (jayahariv) [inline modules/swift/examples/code_snippets/SampleCodeTest.swift]: same like above, can we use simple config initialization?

File: modules/objc/examples/code_snippets/SampleCodeTest.m (hunk 1 above)
Comment (ibsoln) [reply modules/objc/examples/code_snippets/SampleCodeTest.m]: @jayahariv  - see https://github.com/couchbase/docs-couchbase-lite/pull/401/commits/404291a807a98e6671c65d5000d66e147d06fa18

File: modules/swift/examples/code_snippets/SampleCodeTest.swift (hunk 2 above)
Comment (ibsoln) [reply modules/swift/examples/code_snippets/SampleCodeTest.swift]: @jayahariv see https://github.com/couchbase/docs-couchbase-lite/pull/401/commits/109d058b0d5750d2cb325ae9bbccc5edffd527fe

File: modules/objc/examples/code_snippets/SampleCodeTest.m [hunk 3]
```diff
@@ -934,6 +934,25 @@ - (void) dontTestReplicationPullFilter {
...
+//  Added 2/Feb/21 - Ian Bridge
+    - void dontTestCustomHeartbeat {
+    // tag::replication-set-heartbeat[]
+        id target =
+          [[CBLURLEndpoint alloc] initWithURL: [NSURL URLWithString: @"ws://foo.cbl.com/db"]];
+
+        CBLReplicatorConfiguration* config = 
+            [[CBLReplicatorConfiguration alloc] initWithDatabase: db target: target];
+        config.type = kCBLReplicatorTypePush;
+        config.continuous: YES];
```
Comment (jayahariv) [inline modules/objc/examples/code_snippets/SampleCodeTest.m]:
```suggestion
        config.continuous: YES;
```

Comment (jayahariv) [reply modules/objc/examples/code_snippets/SampleCodeTest.m]: do you think, for consistency should we need to add comment `//  other config as required . . .` like Java and .NET?

**Review by jayahariv — APPROVED** [review]

File: modules/csharp/examples/code_snippets/Program.cs [hunk 4]
```diff
@@ -1182,6 +1182,26 @@ private static void PullWithFilter(Database database)
             // end::replication-pull-filter[]
         }
 
+        public void TestReplicatorHeartbeatGetSet()
```
Comment (Sandychuang8) [inline modules/csharp/examples/code_snippets/Program.cs]:
```suggestion
        public void TestCustomHeartbeat()
```
Change test method name to match iOS and java/android.

**Review by Sandychuang8 — APPROVED** [review]
Just method naming change to match with other platforms. Other than that, .Net section looks good.

**Comment by ibsoln** [comment]
Closed -- approved change now merged into feature branch Feature/DOC-7949-284-Documentation -- https://github.com/couchbase/docs-couchbase-lite/pull/407

**Pull request closed without merging**